Sources/Fuzzilli/Compiler/Compiler.swift

```diff
@@ -412,36 +412,69 @@ public class JavaScriptCompiler {
             }
 
         case .forInLoop(let forInLoop):
-            let initializer = forInLoop.left;
-            guard !initializer.hasValue else {
-                throw CompilerError.invalidNodeError("Expected no initial value for the variable declared in a for-in loop")
-            }
-
+            let initializer = forInLoop.left!
             let obj = try compileExpression(forInLoop.right)
+            // Processing a for-in or for-of loop requires an iterator, which is typically declared in the function header.
+            // Alternatively, an existing variable can be used, resulting in an identifier instead of a variable declarator.
+            // If the identifier is not previously declared, it is implicitly created as a global variable.
+
+            switch initializer {
+            case .variableDeclarator(let variableDeclarator):
+                guard !variableDeclarator.hasValue else {
+                    throw CompilerError.invalidNodeError("Expected no initial value for the variable declared in a for-in loop")
+                } 
+                let loopVar = emit(BeginPlainForInLoop(), withInputs: [obj]).innerOutput
+                try enterNewScope {
+                    map(variableDeclarator.name, to: loopVar)
+                    try compileBody(forInLoop.body)
+                }
+                emit(EndForInLoop())
 
-            let loopVar = emit(BeginForInLoop(), withInputs: [obj]).innerOutput
-            try enterNewScope {
-                map(initializer.name, to: loopVar)
-                try compileBody(forInLoop.body)
+            case .identifier(let identifier):
+                let loopVar: Variable
+                if let existingVar = lookupIdentifier(identifier.name) {
+                    loopVar = existingVar
+                } else {
+                    loopVar = emit(LoadNamedVariable(identifier.name)).output
+                    map(identifier.name, to: loopVar)
```

Comment:
Because the case starting in line 438 is used for global variables and the case in line 436 for non-global variables. In both cases, the compiler just has an identfier as input. For global variables, lookupIdentifer will return **nil** because the identifier is not associted with a FuzzIL variable. Either because the global Variable has not been set but also when it has actually been set before, i.e. with StoreNamedVariable (see example "a" and "b")

## Illustration
```javascript
for (a of ["new a"]) {} // line 469. lookupIdentifier didn't return FuzzIL Variable because a is used for the first time here
output("value of a: " + a);

b = "old b";
for (b of ["new b"]) {} // line 469. lookupIdentifier didn't return FuzzIL Variable because the b identifier was not mapped to a FuzzIL variable
output("value of b: " + b);

var c = "old c";
for (c of ["new c"]) {} // line 467 because lookupIdentifier returned the FuzzIL Variable that the previous line mapped the identifier 'c' to
output("value of c: " + c);

for (var d of ["new d"]) {} // line 457. lhs of for loop header is not an identfier...
output("value of d: " + d);
```